sled-agent/src/bootstrap/agent/rack_ops.rs

```diff
@@ -0,0 +1,300 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Internal API for rack-level bootstrap agent operations.
+
+use super::Agent;
+use crate::bootstrap::http_entrypoints::RackOperationStatus;
+use crate::bootstrap::params::RackInitializeRequest;
+use crate::bootstrap::rss_handle::RssHandle;
+use crate::config::SidecarRevision;
+use crate::rack_setup::service::SetupServiceError;
+use std::mem;
+use std::sync::Arc;
+use std::sync::Mutex;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::error::TryRecvError;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RackInitId(pub Uuid);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RackResetId(pub Uuid);
+
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum ConcurrentRssAccess {
```

Comment:
Could we call this something like `RssAccessError`? The `Error` part is more salient than the `Concurrency` part - it's possible a single client making sequential requests triggers one of these.

Reply:
Renamed in [5f71e54](https://github.com/oxidecomputer/omicron/pull/3351/commits/5f71e543309f98da4a2bbc72ab83a8dce7823f6b)